pkg/logql/log/parser_hints.go

```diff
@@ -57,30 +72,52 @@ func (p *parserHint) ShouldExtractPrefix(prefix string) bool {
 	return false
 }
 
-func (p *parserHint) NoLabels() bool {
-	return p.noLabels
+func (p *Hints) NoLabels() bool {
+	return p.noLabels || p.AllRequiredExtracted()
+}
+
+func (p *Hints) RecordExtracted(key string) {
+	for _, l := range p.requiredLabels {
+		if l == key {
+			p.extracted = append(p.extracted, key)
+			return
+		}
+	}
+}
+
+func (p *Hints) AllRequiredExtracted() bool {
+	if len(p.requiredLabels) == 0 {
+		return false
```

Comment:
Isn't this true? If not labels need to be extracted all are already extracted right?